src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs

```diff
@@ -36,24 +37,44 @@ public partial class UpdateDurableFunction
     /// </exception>
     [FunctionName(nameof(UpdateInstanceWatermarkAsync))]
     public async Task<IReadOnlyList<InstanceFileState>> UpdateInstanceWatermarkAsync([ActivityTrigger] UpdateInstanceWatermarkArguments arguments, ILogger logger)
+    {
+        EnsureArg.IsNotNull(arguments, nameof(arguments));
+        var metadatas = await UpdateInstanceWatermarkV2Async(
+            new UpdateInstanceWatermarkArguments(new Partition(arguments.PartitionKey, Partition.UnknownName), arguments.StudyInstanceUid),
+            logger);
+        return metadatas.Select(x => new InstanceFileState
+        {
+            Version = x.VersionedInstanceIdentifier.Version,
+            OriginalVersion = x.InstanceProperties.OriginalVersion,
+            NewVersion = x.InstanceProperties.NewVersion
+        }).ToList();
```

Comment:
You don't actually need to make this a list. You can return `IEnumerable` because this will be serialized to JSON which will be an array `[ { ... }, { ... } ]` regardless of the .NET collection type.